### PR TITLE
[TIMOB-23836] iOS10: Support for table-view prefetching

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23836

### CODE REVIEW

File: iphone/Classes/TiUIListView.m [hunk 1]
```diff
@@ -97,7 +97,14 @@ - (void)dealloc
...
     _tableView.dataSource = nil;
-    [_tableView release];
+
+#if IS_XCODE_8
+    if ([TiUtils isIOS10OrGreater]) {
+        _tableView.prefetchDataSource = nil;
+    }
+#endif
+
+        [_tableView release];
```
Comment (cheekiatng) [inline iphone/Classes/TiUIListView.m]: indentation

File: iphone/Classes/TiUIListView.m [hunk 2]
```diff
@@ -1492,6 +1505,65 @@ - (NSString *)tableView:(UITableView *)tableView titleForFooterInSection:(NSInte
...
+}
+#endif
+
+- (NSDictionary*)listItemFromIndexPath:(NSIndexPath*)indexPath
+{
+    NSIndexPath *realIndexPath = [self pathForSearchPath:indexPath];
+    
+    TiUIListSectionProxy *section = [self.listViewProxy sectionForIndex:realIndexPath.section];
+    NSDictionary *item = [section itemAtIndex:realIndexPath.row];
+    NSMutableDictionary *eventObject = [[NSMutableDictionary alloc] initWithObjectsAndKeys:
```
Comment (cheekiatng) [inline iphone/Classes/TiUIListView.m]: doing alloc here means we have to release it eventually. maybe add `autorelease`.
OR, do `dictionaryWithCapacity` instead, then set each key and object.

Comment (hansemannn) [inline iphone/Classes/TiUIListView.m]: Applied the `autorelease` fix, because the capacity can vary with `itemId`.

**Comment by cheekiatng** [comment]
APPROVED!

**Pull request merged**